backend/app/schemas/schedule.py

```diff
@@ -0,0 +1,52 @@
+from enum import Enum
+from uuid import UUID
+from datetime import datetime, timedelta
+from typing import List, Optional
+from app.schemas.time_block import TimeBlockBase, TimeBlockId, TimeBlockFull
+from pydantic import BaseModel, ConfigDict
+
+
+
+class ScheduleState(str, Enum):
+    PENDING_PARTICIPANT = "PENDING_PARTICIPANT_RESPONSE"
+    PENDING_VOLUNTEER = "PENDING_VOLUNTEER_RESPONSE"
+    SCHEDULED = "SCHEDULED"
+    COMPLETED = "COMPLETED"
+
+    @classmethod
+    def to_schedule_state_id(cls, state: "ScheduleState") -> int:
+        state_map = {
+            cls.PENDING_VOLUNTEER: 1,
+            cls.PENDING_PARTICIPANT: 2,
+            cls.SCHEDULED: 3,
+            cls.COMPLETED: 4}
+
+        return state_map[state]
+
+class ScheduleBase(BaseModel):
+    scheduled_time: Optional[datetime]
+    duration: Optional[timedelta]
+    state_id: int
+
+
+class ScheduleInDB(ScheduleBase):
```

Comment:
I would call this `ScheduleResponse` or `ScheduleCreateResponse`, depending if you think it will be unique to the CREATE schedule request. the `Schedule` object is already a representation of the schedule in the DB, and we don't always want a full representation of what's in the DB in our output, so I don't want to signal that, eg. some places just output id instead of id properties in their API responses.

Reply:
We could also name this ScheduleEntity and then output that if that makes more sense.